core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -684,7 +700,7 @@ class ProducerStateManager(
     }
 
     if (logEndOffset != mapEndOffset) {
-      producers.clear()
```

Comment:
Maybe we can have one more helper for clear so we don't need to remove one by one?

Reply:
I considered another helper, but `clear()` iterates all the keys anyways so I thought it would be cleaner this way
```
def clear(): Unit = { keysIterator foreach -= }
```

Reply:
Oh I see. Thanks for teaching me something :)